src/app/loadout-drawer/loadout-apply.ts

```diff
@@ -249,17 +251,37 @@ function doApplyLoadout(
 
       // Figure out which items have specific socket overrides that will need to be applied.
       // TODO: remove socket-overrides from the mods to apply list!
-      const itemsWithOverrides = loadout.items.filter((loadoutItem) => {
-        const item = getLoadoutItem(loadoutItem);
-        return (
-          loadoutItem.socketOverrides &&
-          item &&
-          // Don't apply perks/mods/subclass configs when moving items to the vault
-          !store.isVault &&
-          // Only apply perks/mods/subclass configs if the item is usable by the store we're applying to
-          (item.classType === DestinyClass.Unknown || item.classType === store.classType)
-        );
-      });
+      const itemsWithOverrides = _.compact(
+        loadout.items.map((loadoutItem) => {
+          const item = getLoadoutItem(loadoutItem);
+          if (
+            !loadoutItem.socketOverrides ||
```

Comment:
Do we currently or have plans to use socket overrides in items other than subclasses? Last I checked it was only subclasses. Considering it has been a year it makes me think that maybe we should be simplifying this somehow so socket overrides are just a property of subclass items only. Maybe something like our `PluggableInventoryItemDefinition`.

Reply:
seems like we will need this for weapons really soon, right?

Reply:
News to me, will ping you on slack about it